profiles/source_profile.go

```diff
@@ -483,11 +494,19 @@ type ShardConfigurationBulk struct {
 type ShardConfigurationDMS struct {
 }
 
+// TODO: Define the sharding structure for dataproc migrations here.
+type ShardConfigurationDataproc struct {
+	SchemaSource DirectConnectionConfig `json:"schemaSource"`
+	// DataShards     []*DataShard           `json:"dataShards"`
```

Comment:
Even if we don't do the sharding in this PR, please ensure the structure is compatible with sharded migrations in the future.

Reply:
`ShardConfigurationDataproc` will be compatible with sharding in future, only `DataShards` will have to be uncommented (no changes will be required in existing struct members).

Had it commented as it's being in use anywhere, can comment right now itself, so that it'll be auto initialized to null or empty array

Let me know, what you think is better.